lib/full-site-editing/block-templates.php

```diff
@@ -105,6 +105,42 @@ function _gutenberg_get_template_files( $template_type ) {
 	return $template_files;
 }
 
+/**
+ * Parses wp_template content and injects the current theme's
+ * stylesheet as a theme attribute into each wp_template_part
+ *
+ * @param string $template_content serialized wp_template content.
+ * @param string $theme the active theme's stylesheet.
+ *
+ * @return string Updated wp_template content.
+ */
+function _inject_theme_attribute_in_content( $template_content, $theme ) {
+	$has_updated_content = false;
+	$new_content         = '';
+	$template_blocks     = parse_blocks( $template_content );
+
+	foreach ( $template_blocks as $key => $block ) {
+		if (
+			'core/template-part' === $block['blockName'] &&
+			! isset( $block['attrs']['theme'] ) &&
+			wp_get_theme()->get_stylesheet() === $theme
+		) {
+			$template_blocks[ $key ]['attrs']['theme'] = $theme;
+			$has_updated_content                       = true;
+		}
+	}
+
+	if ( $has_updated_content ) {
+		foreach ( $template_blocks as $block ) {
+			$new_content .= serialize_block( $block );
+		}
+
+		return $new_content;
+	} else {
```

Comment:
Nitpick: we can drop the else block here since if ends with return, and just `return $template_content` at top level.

Reply:
Sure! Not strongly opinionated about early returns. If it's a convention, my mistake! I thought `run composer format` would catch it.

Reply:
It might not be a written one but it's less keywords and nesting that achieves the same effect. 😄  Also that way it's immediately obvious that this function has a default return value while quickly scanning the code, at least to me.